runtime/interpreter/value.go

```diff
@@ -16709,7 +16841,7 @@ func (v AddressValue) Transfer(
 }
 
 func (v AddressValue) Clone(_ *Interpreter) Value {
-	return v
+	return NewUnmeteredAddressValue(v[:])
```

Comment:
leaving this as is should be ok. so we won't allocate extra memory (new slices) during internal operation such as migrations, etc.

Reply:
It's weird to me that that our `Clone` methods return the original most of the time. For e.g. `BoolValue` that doesn't matter because changing the returned value won't change the original. But for slices like `AddressValue`, changes to the copy affect the original.

I suppose this is a naming argument. This isn't the right PR for that argument though so I'll revert it.